### PR TITLE
How to use local storage with JavaScript

### LINKED ISSUE #802: How to use local storage with JavaScript
### Brief Summary: 

Local storage allow web applications to store temporary data on the browser. This information helps to improves the site's functionality. For instance, the data can help track users as they navigate across different pages. Local storage is cleared once the user closes the web page.

### Key Takeaways: 

 - What is local storage
 - How to perform CRUD functionalities on local storage
 - Debug errors showing up in the console.

### References: 

https://javascript.info/localstorage

### PR DESCRIPTION
**Pull request by WanjaMIKE**
This fixes #802

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by hectorkambow** [comment]
Plagiarism check done

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:48]: Rewrite as

This function is used to access or ....

Comment (zolomohan) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:48]: It then extracts the required value. Can you be more clear?

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:62]: Remove `To make maximum use of the above result,`.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:76]: Rewrite as

This method is used to clear all values in the localStorage.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:82]: Remove `&#39;`. Use normal apostrophes.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:85]: The heading must be named as Prerequisites.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:90]: Prerequisites and Goals must be at the top of the article after the introduction.

Comment (zolomohan) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:90]: You can add an overview if you want to give a brief intro about the topics and steps you will cover in this article.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:85]: Provide download links for Visual Studio Code and Chrome.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:93]: These methods will be used to access different functionalities of the `localStorage`.

You must say functions, not methods,

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:219]: The functions aren't clickable. The buttons are. 
To be more accurate,
You must say the `onclick` property of the buttons will be set only after the page has loaded.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:242]: These sections should be at the top after you explain the Local Storage methods, or before it.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:245]: `In case, you have not understood any concept,`

Replace have not with didn't

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:248]: Fix Infor. It's incomplete.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:217]: can you briefly explain the other methods that you are using like document.createElement and appendChild etc.

Break the four functions into separate code snippets and briefly explain about them.

**Review by zolomohan — CHANGES_REQUESTED** [review]
@WanjaMIKE 
Overall, A good article.

I've left some suggestions regarding the flow of the article and some minor grammatical errors.
Ping me when you have implemented them.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (WanjaMIKE) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:1]: its localStorage not Local Storage. This cannot be changed. Let's in md is writen as let&#39;s. so is browser's. So this syntax is correct.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (WanjaMIKE) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:4]: Revised

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (WanjaMIKE) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:15]: that is a comment in line. the method is already explained as shown. Writing outside the code snippet is repetition.

#### key()   ---method explained
This function is used to retrieve a value/string in a specific location. The index can be passed into the key() function as a parameter.

```javascript
var answer = localStorage.key(1);
// this statement will retrieve the value of the second item in the localStorage.
```

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (WanjaMIKE) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:18]: Revised

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (WanjaMIKE) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:219]: the functions will only be accessible after the page has finished loading..... - Revised

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (WanjaMIKE) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:242]: Revised

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (WanjaMIKE) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:245]: Revised

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (WanjaMIKE) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:248]: Actually the website is called JavaScript.Infor   - Revised

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (WanjaMIKE) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:217]: Revised

**Comment by WanjaMIKE** [comment]
@zolomohan

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:20]: hmm, fine.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [reply articles/how-to-use-local-storage-with-JavaScript/index.md]: localStorage is the object name in javascript. Whereas Local Storage is the topic you are discussing here. If you want to use localstorage, please mark it like this. `localStorage`.

I still suggest using Local Storage.

Comment (zolomohan) [reply articles/how-to-use-local-storage-with-JavaScript/index.md]: image

Even in the developer tools, it's referenced as Local Storage. not localStorage.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:34]: you can say, As mentioned before.

**Comment by zolomohan** [comment]
Hi @WanjaMIKE, I noticed that all the suggestions were not implemented. Please go through them once again. I resolved all the conversations that were done. So please go through the unresolved ones.

**Comment by WanjaMIKE** [comment]
Hey @zolomohan 
Hey, I made all changes apart from this `Remove `&#39;` As I commented that is correct.

**Comment by WanjaMIKE** [comment]
```&#39;``` removed

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [reply articles/how-to-use-local-storage-with-JavaScript/index.md]: @WanjaMIKE, This is not done yet.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:38]: `Window` is wrong. It's `window`. JS is case sensitive. Please fix this everywhere in the article.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:107]: You've added a Bootsrap link here, but you didn't instruct the readers to add a bootstrap file.

I think it's better if you provide a bootstrap cdn link here.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:129]: This navbar isn't related to local storage. 

Remove this. We must keep the code very simple for the readers to understand easily.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:132]: Give a proper heading in this `h1`. Like Local Storage or something.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:137]: Update the button text to Store Records. To be consistent with the other buttons.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [reply articles/how-to-use-local-storage-with-JavaScript/index.md:107]: Also, inform the readers that you are using bootstrap and that is causing the styling in the webpage. 

And provide a link to Bootstrap's homepage.

File: articles/how-to-use-local-storage-with-JavaScript/index.md
Comment (zolomohan) [inline articles/how-to-use-local-storage-with-JavaScript/index.md:281]: Update the conclusion with each topic you covered.

**Review by zolomohan — CHANGES_REQUESTED** [review]
@WanjaMIKE 
Please fix these and we'll move to a final review.

**Review by zolomohan — APPROVED** [review]

**Review by louisefindlay23 — APPROVED** [review]
This article has passed final review, @hectorkambow.

Pending publication till @zolomohan makes local storage/session storage case consistent between this and @WanjaMIKE's related article #1045.

**Review by zolomohan — APPROVED** [review]

**Comment by zolomohan** [comment]
Fixed the @louisefindlay23

**Comment by louisefindlay23** [comment]
Ready for publication now, @hectorkambow.

**Pull request merged**
